Upgrade test is flaky

### What happened?

https://prow.k8s.io/?job=periodic-agent-sandbox-migration-test

Test flakes look similar:

```
Traceback (most recent call last):
  File "/home/prow/go/src/github.com/kubernetes-sigs/agent-sandbox/dev/tools/test-migration.py", line 851, in <module>
    main()
  File "/home/prow/go/src/github.com/kubernetes-sigs/agent-sandbox/dev/tools/test-migration.py", line 822, in main
    active_pod_info = create_v1alpha1_objects()
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/prow/go/src/github.com/kubernetes-sigs/agent-sandbox/dev/tools/test-migration.py", line 315, in create_v1alpha1_objects
    assert pod_exists, "Pod upgrade-sandbox-running was not created in time!"
AssertionError: Pod upgrade-sandbox-running was not created in time!
```

### How can we reproduce it (as minimally and precisely as possible)?

See above

### Version

HEAD

### Anything else we need to know?

_No response_

--- a/dev/tools/test-migration.py
+++ b/dev/tools/test-migration.py
@@ -305,12 +305,20 @@ def create_v1alpha1_objects():
     
     print("Waiting for upgrade-sandbox-running Pod to be created...")
     pod_exists = False
-    for i in range(20):
+    for i in range(60):
         res = run_cmd(["kubectl", "get", "pod", "upgrade-sandbox-running", "-n", "default"], check=False, capture_output=True)
         if res.returncode == 0:
             pod_exists = True
             break
-        time.sleep(1)
+        if (i + 1) % 5 == 0:
+            print(f"Pod not created yet (attempt {i+1}/60)...")
+        time.sleep(2)
+        
+    if not pod_exists:
+        print("Pod upgrade-sandbox-running was not created in time. Dumping diagnostics:", file=sys.stderr)
+        run_cmd(["kubectl", "get", "sandboxes,pods", "-n", "default"], check=False)
+        run_cmd(["kubectl", "describe", "sandbox", "upgrade-sandbox-running", "-n", "default"], check=False)
+        run_cmd(["kubectl", "logs", "-n", "agent-sandbox-system", "deploy/agent-sandbox-controller", "--tail=100"], check=False)
         
     assert pod_exists, "Pod upgrade-sandbox-running was not created in time!"
     

